### DISCUSSION TITLE
Using Planetary Computer for web app backend to replace Google Earth Engine

### DISCUSSION THREAD

**Discussion by staffordsmith83** [microsoft/PlanetaryComputer#297] [General]
I have been building web applications using Googel Earth Engine as a geospatial processing backend. I make calls to GEE using the [JS API](https://github.com/google/earthengine-api), perform geospatial analysis on Earth Engine assets (including my own data that I have uploaded), and return the map results as tiled map services to my front end application.

Does Planetary Computer work like this?

With GEE I can easily perform things like [image segmentation](https://developers.google.com/earth-engine/apidocs/ee-algorithms-image-segmentation-kmeans) using google compute. I just call the operation from my front end by specifying algorithm parameters and datasets, and get a result to display in my front end.

I would love to replace this with Planetary Computer instead!

**Comment by TomAugspurger** [comment]
> Does Planetary Computer work like this?

Not exactly the same, but you can achieve parts of what you're trying to do.

Our STAC API returns URLs directly to blob storage, which you can read using any program that can make HTTP requests, so you can choose whatever kind of compute layer you want.

If you're looking for a setup where you don't have *any* compute layer, you might want to look through our [data API](https://planetarycomputer.microsoft.com/docs/quickstarts/using-the-data-api/). That, along with our STAC API, powers our [explorer](https://planetarycomputer.microsoft.com/docs/overview/explorer/). But you'll see that the data API is primarily focused on tiling and visualization.

**Reply by staffordsmith83** [reply]
Ah yes I saw the data API, but has limited geoprocessing capability. Still a great product, but I suppose not a direct replacement for Earth Engine if there is no included compute. A pity, I dont like the strange GEE way of doing things!

**Comment by gilbertocamara** [comment]
Dear @staffordsmith83 @TomAugspurger Please check the R `sits` package, which provides an end-to-end environment for land use and land cover change classification. `sits` has many functionalities which are not provided in GEE, including: (a) advanced methods for improving the quality of training samples; (b) deep learning methods for image classification; (c) Bayesian methods for post-processing; (d) support for uncertainty mapping and active learning; (e) object-based image time series analysis; (f) STAC based access to Planetary Computer collections.

**Comment by gilbertocamara** [comment]
Dear @staffordsmith83 Thanks to the excellent support provided by @TomAugspurger, `sits` is available in MPC hub, in its R environment.
